### DISCUSSION TITLE
Adding yaw rate

### DISCUSSION THREAD

**Discussion by wrgoto** [snipem/gt7dashboard#14] [Ideas]
Hello! Thanks for this amazing tool. Any chance you can add yaw rate to the telemetry?

Visualizing how much the car is rotating during corners really helps understand where improvements can be made.

**Comment by snipem** [comment]
Interesting you asking this. I already implemented yaw rate in the improvements branch. You may want to take a look: https://github.com/snipem/gt7dashboard/tree/improvements 

I implemented it as "Yaw Rate per second". With the yaw angle normalised to positive numbers regardless of direction of yaw. When ever the Yaw Rate peaks, throttle should be engaged. At least that is what I would be looking at.

![image](https://github.com/snipem/gt7dashboard/assets/985581/cfdf7a38-b827-4ce3-84a3-b708f9970d35)

**Reply by wrgoto** [reply]
Nice! I'll take a look. FYI, using the latest commit on `main` causes my client to duplicate laps. If I run like 5 laps, it logs like ~200 laps and they're all duplicates.

**Reply by snipem** [reply]
I reverted the changes I made to the main branch. This should make the main branch functional identical with the 1.1 release. My newest changes are still in the "improvements" branch if you are interested.

Can you check if this solves your issue?

**Reply by wrgoto** [reply]
Solved the issue on main, but the issue still exists on the improvements branch. Thanks again for working on this, it's been helping me find so much time when I prep for my league races.

**Reply by snipem** [reply]
Nice. Can you give me some examples what data you use and in what way to improve? Trying to improve myself.

**Reply by wrgoto** [reply]
Not sure if you misread my comment above, but the issue you solved on `main` still exists on the `improvements` branch.

**Reply by snipem** [reply]
I did not misread your comment. I was just curious how you use the tool.

Meanwhile, I hope I fixed the issue with this commit 4b4fd753

**Reply by wrgoto** [reply]
Sorry, I didn't see that you acknowledge that the bug existed on `improvements`, I'll check it out in a bit.

I use the telemetry in a number of ways. First, I need to make sure that my braking, maximum rotation point, and acceleration are spot on with the curve and the apex of the turn, so I look to see if my max braking is at the beginning of the braking zone it trails off into the apex, all the while my rotation is increasing until I hit the apex. Then I can see if I'm on the accelerator right after the apex.

After a few laps, I'll compare time gained/lossed against my last lap with the fastest while trying out different lines etc to see if there's any time I can gain in the lap. Pretty standard I feel.
